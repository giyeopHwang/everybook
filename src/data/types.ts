export enum TagTypes {
  on,
  off,
}

export type NavBarCardDetail = {
  name: string;
  amount: number;
  link?: string;
};

export type NavBarNet = {
  title: string;
  amount: number;
  type: 'net';
};

export type NavBarPlus = {
  title: string;
  amount: number;
  type: 'plus';
  details: NavBarCardDetail[];
};

export type NavBarMinus = {
  title: string;
  amount: number;
  type: 'minus';
  details: NavBarCardDetail[];
};
