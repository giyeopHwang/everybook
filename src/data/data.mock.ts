import dayjs from 'dayjs';
import { ColDef } from 'ag-grid-community';
import { Expense } from '@/types/types';
import { suppressKeyboardEvent } from '@/components/worksheet/expense-panel/expense-panel.utils';

export const mockExpenseRowData: Expense[] = [
  {
    id: '1',
    isSelected: false,
    date: new Date('2024.01.01'),
    place: '다이소',
    item: '주방세제',
    cash: 2_000,
    card: 0,
    cashAccount: 'PAYCO',
    cardAccount: null,
    category: '생활용품>주방/욕실',
    isWaste: false,
  },
  {
    id: '2',
    isSelected: false,
    date: new Date('2024.01.01'),
    place: '서울호떡',
    item: '호떡',
    cash: 3_000,
    card: 0,
    cashAccount: '네이버페이머니통장',
    cardAccount: '네이버페이체크카드',
    category: '식비>간식',
    isWaste: false,
  },
  {
    id: '3',
    isSelected: false,
    date: new Date('2024.01.01'),
    place: '농부의아들',
    item: '고구마',
    cash: 3_000,
    card: 0,
    cashAccount: '네이버페이머니통장',
    cardAccount: '네이버페이체크카드',
    category: '식비>간식',
    isWaste: false,
  },
  {
    id: '4',
    isSelected: false,
    date: new Date('2024.01.02'),
    place: '롯데리아',
    item: '핫크리스피버거세트',
    cash: 0,
    card: 7_800,
    cashAccount: null,
    cardAccount: '톡톡 my point 카드',
    category: '식비>외식',
    isWaste: true,
  },
  {
    id: '5',
    isSelected: false,
    date: new Date('2024.01.02'),
    place: '쿠팡',
    item: '에어프라이어',
    cash: 0,
    card: 150_000,
    cashAccount: null,
    cardAccount: '톡톡 my point 카드',
    category: '생활용품>가구/가전',
    isWaste: false,
  },
  {
    id: '6',
    isSelected: false,
    date: new Date('2024.01.03'),
    place: '국민건강보험',
    item: '건강보험료',
    cash: 45_000,
    card: 0,
    cashAccount: '하나주거래통장',
    cardAccount: null,
    category: '건강/문화>보장성보험',
    isWaste: false,
  },
  {
    id: '7',
    isSelected: false,
    date: new Date('2024.01.03'),
    place: '어머니',
    item: '용돈',
    cash: 200_000,
    card: 0,
    cashAccount: '하나주거래통장',
    cardAccount: null,
    category: '용돈/기타>용돈',
    isWaste: false,
  },
  {
    id: '8',
    isSelected: false,
    date: new Date('2024.01.04'),
    place: 'GAK',
    item: '커트',
    cash: 23_000,
    card: 0,
    cashAccount: '네이버페이머니통장',
    cardAccount: '네이버페이체크카드',
    category: '의복/미용>헤어/뷰티',
    isWaste: false,
  },
  {
    id: '9',
    isSelected: false,
    date: new Date('2024.01.04'),
    place: '농부의아들',
    item: '귤',
    cash: 5_000,
    card: 0,
    cashAccount: '네이버페이머니통장',
    cardAccount: '네이버페이체크카드',
    category: '식비>간식',
    isWaste: false,
  },
  {
    id: '10',
    isSelected: false,
    date: new Date('2024.01.05'),
    place: '전통한우푸줏간',
    item: '삼겹살',
    cash: 0,
    card: 13_000,
    cashAccount: null,
    cardAccount: '현대카드ZERO',
    category: '식비>주식',
    isWaste: false,
  },
  {
    id: '11',
    isSelected: false,
    date: new Date('2024.01.05'),
    place: 'GS25',
    item: '과자',
    cash: 3_000,
    card: 0,
    cashAccount: 'PAYCO',
    cardAccount: null,
    category: '식비>간식',
    isWaste: true,
  },
  {
    id: '12',
    isSelected: false,
    date: new Date('2024.01.05'),
    place: '마이프로틴',
    item: '보충제',
    cash: 0,
    card: 70_000,
    cashAccount: null,
    cardAccount: '톡톡 my point 카드',
    category: '건강/문화>기타',
    isWaste: false,
  },
  {
    id: '13',
    isSelected: false,
    date: new Date('2024.01.06'),
    place: '다이소',
    item: '밀폐용기',
    cash: 3_000,
    card: 0,
    cashAccount: 'PAYCO',
    cardAccount: null,
    category: '생활용품>주방/욕실',
    isWaste: true,
  },
  {
    id: '14',
    isSelected: false,
    date: new Date('2024.01.06'),
    place: '인터넷등기소',
    item: '수수료',
    cash: 700,
    card: 0,
    cashAccount: '네이버페이머니통장',
    cardAccount: '네이버페이체크카드',
    category: '주거/통신>기타',
    isWaste: false,
  },
  {
    id: '15',
    isSelected: false,
    date: new Date('2024.01.07'),
    place: '하나은행',
    item: '대출이자',
    cash: 250_000,
    card: 0,
    cashAccount: '하나주거래통장',
    cardAccount: '',
    category: '세금/이자>이자',
    isWaste: false,
  },
];

export const mockExpenseColDefs: ColDef<Expense>[] = [
  { field: 'isSelected', headerName: '선택', editable: true, flex: 0.5 },
  {
    field: 'date',
    headerName: '날짜',
    editable: true,
    valueFormatter: (params) => `${dayjs(params.value).format('YYYY.MM.DD')}`,
    flex: 0.75,
  },
  {
    field: 'place',
    headerName: '사용처',
    editable: true,
    suppressKeyboardEvent,
    flex: 1,
  },
  {
    field: 'item',
    headerName: '사용내역',
    editable: true,
    suppressKeyboardEvent,
    flex: 1,
  },
  {
    field: 'cash',
    headerName: '현금',
    editable: true,
    valueFormatter: (params) =>
      params.value && `₩${params.value.toLocaleString()}`,
    flex: 1,
  },
  {
    field: 'card',
    headerName: '카드',
    editable: true,
    valueFormatter: (params) =>
      params.value && `₩${params.value.toLocaleString()}`,
    flex: 1,
  },
  {
    field: 'cashAccount',
    headerName: '출금통장',
    editable: true,
    cellEditor: 'agSelectCellEditor',
    cellEditorParams: {
      values: ['선택없음', '하나주거래통장', '네이버페이머니통장', 'PAYCO'],
    },
    valueSetter: (params) => {
      params.data.cashAccount =
        params.newValue === '선택없음' ? null : params.newValue;
      return true;
    },
    flex: 1,
  },
  {
    field: 'cardAccount',
    headerName: '카드분류',
    editable: true,
    cellEditor: 'agSelectCellEditor',
    cellEditorParams: {
      values: [
        '선택없음',
        '네이버페이체크카드',
        '톡톡 my point 카드',
        '현대카드ZERO',
      ],
    },
    valueSetter: (params) => {
      params.data.cardAccount =
        params.newValue === '선택없음' ? null : params.newValue;
      return true;
    },
    flex: 1,
  },
  { field: 'category', headerName: '분류', editable: true, flex: 1 },
  {
    field: 'isWaste',
    headerName: '낭비',
    editable: true,
    flex: 0.5,
  },
];
