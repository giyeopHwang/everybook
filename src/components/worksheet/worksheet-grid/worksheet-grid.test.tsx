import { render } from '@testing-library/react';
import WorksheetGrid from './worksheet-grid.component';
import { mockExpenseRowData, mockExpenseColDefs } from '@/data/data.mock';

describe('WorksheetGrid', () => {
  it('renders WorksheetGrid component', () => {
    const mock = vi.fn();
    const worksheetGrid = render(
      <WorksheetGrid
        rowData={mockExpenseRowData}
        colDefs={mockExpenseColDefs}
        onRowDataUpdate={mock}
      />
    );
    expect(worksheetGrid).toMatchSnapshot();
  });
});
