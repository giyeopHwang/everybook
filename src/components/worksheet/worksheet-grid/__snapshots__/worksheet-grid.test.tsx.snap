// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExpenseGrid > renders ExpenseGrid component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ag-theme-quartz css-1522fkl MuiBox-root css-0"
        style="height: 500px;"
      >
        <div
          style="height: 100%;"
        >
          <!-- AG Grid -->
          <div
            class="ag-root-wrapper ag-ltr ag-layout-normal"
            grid-id="1"
            role="presentation"
          >
            <div
              class="ag-root-wrapper-body ag-focus-managed ag-layout-normal"
              role="presentation"
            >
              <div
                class="ag-tab-guard ag-tab-guard-top"
                role="presentation"
                tabindex="0"
              />
              <!-- AG Grid Body -->
              <div
                aria-colcount="0"
                aria-rowcount="0"
                class="ag-root ag-unselectable ag-layout-normal"
                role="treegrid"
              >
                <div
                  class="ag-header ag-pivot-off"
                  role="presentation"
                  style="height: 1px; min-height: 1px;"
                >
                  <div
                    aria-hidden="false"
                    class="ag-pinned-left-header "
                    role="presentation"
                  />
                  <div
                    class="ag-header-viewport "
                    role="presentation"
                  >
                    <div
                      class="ag-header-container"
                      role="rowgroup"
                      style="width: 0px;"
                    />
                  </div>
                  <div
                    aria-hidden="false"
                    class="ag-pinned-right-header "
                    role="presentation"
                  />
                </div>
                <!-- AG Pinned Top -->
                <div
                  class="ag-floating-top"
                  role="presentation"
                  style="height: 0px; min-height: 0; display: none;"
                >
                  <!-- AG Row Container topLeft -->
                  <div
                    class="ag-pinned-left-floating-top"
                    role="presentation"
                  />
                  <!-- AG Row Container topCenter -->
                  <div
                    class="ag-floating-top-viewport"
                    role="presentation"
                  >
                    <div
                      class="ag-floating-top-container"
                      role="presentation"
                      style="width: 0px;"
                    />
                  </div>
                  <!-- AG Row Container topRight -->
                  <div
                    class="ag-pinned-right-floating-top"
                    role="presentation"
                  />
                  <!-- AG Row Container topFullWidth -->
                  <div
                    class="ag-floating-top-full-width-container"
                    role="presentation"
                  />
                </div>
                <div
                  class="ag-body ag-layout-normal"
                  role="presentation"
                >
                  <!-- AG Middle -->
                  <div
                    class="ag-body-viewport ag-row-animation ag-layout-normal"
                    role="presentation"
                  >
                    <!-- AG Row Container left -->
                    <div
                      class="ag-pinned-left-cols-container"
                      role="presentation"
                      style="height: 1px;"
                    />
                    <!-- AG Row Container center -->
                    <div
                      class="ag-center-cols-viewport"
                      role="presentation"
                      style="height: 1px;"
                    >
                      <div
                        class="ag-center-cols-container"
                        role="presentation"
                        style="width: 0px; height: 1px;"
                      />
                    </div>
                    <!-- AG Row Container right -->
                    <div
                      class="ag-pinned-right-cols-container"
                      role="presentation"
                      style="height: 1px;"
                    />
                    <!-- AG Row Container fullWidth -->
                    <div
                      class="ag-full-width-container"
                      role="presentation"
                      style="height: 1px;"
                    />
                  </div>
                  <!-- AG Fake Vertical Scroll -->
                  <div
                    aria-hidden="true"
                    class="ag-body-vertical-scroll"
                  >
                    
            
                    <div
                      class="ag-body-vertical-scroll-viewport"
                      ref="eViewport"
                    >
                      
                
                      <div
                        class="ag-body-vertical-scroll-container"
                        ref="eContainer"
                        style="height: 1px;"
                      />
                      
            
                    </div>
                    
        
                  </div>
                </div>
                <!-- AG Sticky Top -->
                <div
                  class="ag-sticky-top"
                  role="presentation"
                  style="height: 0px; top: 0px; width: 100%;"
                >
                  <!-- AG Row Container stickyTopLeft -->
                  <div
                    class="ag-pinned-left-sticky-top"
                    role="presentation"
                  />
                  <!-- AG Row Container stickyTopCenter -->
                  <div
                    class="ag-sticky-top-viewport"
                    role="presentation"
                  >
                    <div
                      class="ag-sticky-top-container"
                      role="presentation"
                      style="width: 0px;"
                    />
                  </div>
                  <!-- AG Row Container stickyTopRight -->
                  <div
                    class="ag-pinned-right-sticky-top"
                    role="presentation"
                  />
                  <!-- AG Row Container stickyTopFullWidth -->
                  <div
                    class="ag-sticky-top-full-width-container"
                    role="presentation"
                  />
                </div>
                <!-- AG Pinned Bottom -->
                <div
                  class="ag-floating-bottom"
                  role="presentation"
                  style="height: 0px; min-height: 0; display: none;"
                >
                  <!-- AG Row Container bottomLeft -->
                  <div
                    class="ag-pinned-left-floating-bottom"
                    role="presentation"
                  />
                  <!-- AG Row Container bottomCenter -->
                  <div
                    class="ag-floating-bottom-viewport"
                    role="presentation"
                  >
                    <div
                      class="ag-floating-bottom-container"
                      role="presentation"
                      style="width: 0px;"
                    />
                  </div>
                  <!-- AG Row Container bottomRight -->
                  <div
                    class="ag-pinned-right-floating-bottom"
                    role="presentation"
                  />
                  <!-- AG Row Container bottomFullWidth -->
                  <div
                    class="ag-floating-bottom-full-width-container"
                    role="presentation"
                  />
                </div>
                <!-- AG Fake Horizontal Scroll -->
                <div
                  aria-hidden="true"
                  class="ag-body-horizontal-scroll"
                >
                  
            
                  <div
                    class="ag-horizontal-left-spacer"
                    ref="eLeftSpacer"
                    style="width: 0px; max-width: 0px; min-width: 0px;"
                  />
                  
            
                  <div
                    class="ag-body-horizontal-scroll-viewport"
                    ref="eViewport"
                  >
                    
                
                    <div
                      class="ag-body-horizontal-scroll-container"
                      ref="eContainer"
                      style="width: 0px;"
                    />
                    
            
                  </div>
                  
            
                  <div
                    class="ag-horizontal-right-spacer"
                    ref="eRightSpacer"
                    style="width: 0px; max-width: 0px; min-width: 0px;"
                  />
                  
        
                </div>
                <!-- AG Overlay Wrapper -->
                <div
                  aria-hidden="true"
                  class="ag-overlay"
                >
                  
            
                  <div
                    class="ag-overlay-panel"
                  >
                    
                
                    <div
                      class="ag-overlay-wrapper ag-layout-normal ag-overlay-loading-wrapper"
                      ref="eOverlayWrapper"
                    >
                      <span
                        class="ag-overlay-loading-center"
                      >
                        Loading...
                      </span>
                    </div>
                    
            
                  </div>
                  
        
                </div>
              </div>
              <div
                class="ag-tab-guard ag-tab-guard-bottom"
                role="presentation"
                tabindex="0"
              />
            </div>
            <div
              aria-hidden="true"
              class="ag-paging-panel ag-unselectable ag-hidden"
              id="ag-4"
            >
              
                
              <!--AG-PAGE-SIZE-SELECTOR-->
              <span
                class="ag-paging-page-size"
                ref="pageSizeComp"
              />
              
                
              <span
                class="ag-paging-row-summary-panel"
                role="status"
              >
                
                    
                <span
                  class="ag-paging-row-summary-panel-number"
                  id="ag-4-first-row"
                  ref="lbFirstRowOnPage"
                />
                
                    
                <span
                  id="ag-4-to"
                >
                  to
                </span>
                
                    
                <span
                  class="ag-paging-row-summary-panel-number"
                  id="ag-4-last-row"
                  ref="lbLastRowOnPage"
                />
                
                    
                <span
                  id="ag-4-of"
                >
                  of
                </span>
                
                    
                <span
                  class="ag-paging-row-summary-panel-number"
                  id="ag-4-row-count"
                  ref="lbRecordCount"
                />
                
                
              </span>
              
                
              <span
                class="ag-paging-page-summary-panel"
                role="presentation"
              >
                
                    
                <div
                  aria-label="First Page"
                  class="ag-button ag-paging-button"
                  ref="btFirst"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="ag-icon ag-icon-first"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                    
                <div
                  aria-label="Previous Page"
                  class="ag-button ag-paging-button"
                  ref="btPrevious"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="ag-icon ag-icon-previous"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                    
                <span
                  class="ag-paging-description"
                  role="status"
                >
                  
                        
                  <span
                    id="ag-4-start-page"
                  >
                    Page
                  </span>
                  
                        
                  <span
                    class="ag-paging-number"
                    id="ag-4-start-page-number"
                    ref="lbCurrent"
                  />
                  
                        
                  <span
                    id="ag-4-of-page"
                  >
                    of
                  </span>
                  
                        
                  <span
                    class="ag-paging-number"
                    id="ag-4-of-page-number"
                    ref="lbTotal"
                  />
                  
                    
                </span>
                
                    
                <div
                  aria-label="Next Page"
                  class="ag-button ag-paging-button"
                  ref="btNext"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="ag-icon ag-icon-next"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                    
                <div
                  aria-label="Last Page"
                  class="ag-button ag-paging-button"
                  ref="btLast"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="ag-icon ag-icon-last"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                
              </span>
              
            
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ag-theme-quartz css-1522fkl MuiBox-root css-0"
      style="height: 500px;"
    >
      <div
        style="height: 100%;"
      >
        <!-- AG Grid -->
        <div
          class="ag-root-wrapper ag-ltr ag-layout-normal"
          grid-id="1"
          role="presentation"
        >
          <div
            class="ag-root-wrapper-body ag-focus-managed ag-layout-normal"
            role="presentation"
          >
            <div
              class="ag-tab-guard ag-tab-guard-top"
              role="presentation"
              tabindex="0"
            />
            <!-- AG Grid Body -->
            <div
              aria-colcount="0"
              aria-rowcount="0"
              class="ag-root ag-unselectable ag-layout-normal"
              role="treegrid"
            >
              <div
                class="ag-header ag-pivot-off"
                role="presentation"
                style="height: 1px; min-height: 1px;"
              >
                <div
                  aria-hidden="false"
                  class="ag-pinned-left-header "
                  role="presentation"
                />
                <div
                  class="ag-header-viewport "
                  role="presentation"
                >
                  <div
                    class="ag-header-container"
                    role="rowgroup"
                    style="width: 0px;"
                  />
                </div>
                <div
                  aria-hidden="false"
                  class="ag-pinned-right-header "
                  role="presentation"
                />
              </div>
              <!-- AG Pinned Top -->
              <div
                class="ag-floating-top"
                role="presentation"
                style="height: 0px; min-height: 0; display: none;"
              >
                <!-- AG Row Container topLeft -->
                <div
                  class="ag-pinned-left-floating-top"
                  role="presentation"
                />
                <!-- AG Row Container topCenter -->
                <div
                  class="ag-floating-top-viewport"
                  role="presentation"
                >
                  <div
                    class="ag-floating-top-container"
                    role="presentation"
                    style="width: 0px;"
                  />
                </div>
                <!-- AG Row Container topRight -->
                <div
                  class="ag-pinned-right-floating-top"
                  role="presentation"
                />
                <!-- AG Row Container topFullWidth -->
                <div
                  class="ag-floating-top-full-width-container"
                  role="presentation"
                />
              </div>
              <div
                class="ag-body ag-layout-normal"
                role="presentation"
              >
                <!-- AG Middle -->
                <div
                  class="ag-body-viewport ag-row-animation ag-layout-normal"
                  role="presentation"
                >
                  <!-- AG Row Container left -->
                  <div
                    class="ag-pinned-left-cols-container"
                    role="presentation"
                    style="height: 1px;"
                  />
                  <!-- AG Row Container center -->
                  <div
                    class="ag-center-cols-viewport"
                    role="presentation"
                    style="height: 1px;"
                  >
                    <div
                      class="ag-center-cols-container"
                      role="presentation"
                      style="width: 0px; height: 1px;"
                    />
                  </div>
                  <!-- AG Row Container right -->
                  <div
                    class="ag-pinned-right-cols-container"
                    role="presentation"
                    style="height: 1px;"
                  />
                  <!-- AG Row Container fullWidth -->
                  <div
                    class="ag-full-width-container"
                    role="presentation"
                    style="height: 1px;"
                  />
                </div>
                <!-- AG Fake Vertical Scroll -->
                <div
                  aria-hidden="true"
                  class="ag-body-vertical-scroll"
                >
                  
            
                  <div
                    class="ag-body-vertical-scroll-viewport"
                    ref="eViewport"
                  >
                    
                
                    <div
                      class="ag-body-vertical-scroll-container"
                      ref="eContainer"
                      style="height: 1px;"
                    />
                    
            
                  </div>
                  
        
                </div>
              </div>
              <!-- AG Sticky Top -->
              <div
                class="ag-sticky-top"
                role="presentation"
                style="height: 0px; top: 0px; width: 100%;"
              >
                <!-- AG Row Container stickyTopLeft -->
                <div
                  class="ag-pinned-left-sticky-top"
                  role="presentation"
                />
                <!-- AG Row Container stickyTopCenter -->
                <div
                  class="ag-sticky-top-viewport"
                  role="presentation"
                >
                  <div
                    class="ag-sticky-top-container"
                    role="presentation"
                    style="width: 0px;"
                  />
                </div>
                <!-- AG Row Container stickyTopRight -->
                <div
                  class="ag-pinned-right-sticky-top"
                  role="presentation"
                />
                <!-- AG Row Container stickyTopFullWidth -->
                <div
                  class="ag-sticky-top-full-width-container"
                  role="presentation"
                />
              </div>
              <!-- AG Pinned Bottom -->
              <div
                class="ag-floating-bottom"
                role="presentation"
                style="height: 0px; min-height: 0; display: none;"
              >
                <!-- AG Row Container bottomLeft -->
                <div
                  class="ag-pinned-left-floating-bottom"
                  role="presentation"
                />
                <!-- AG Row Container bottomCenter -->
                <div
                  class="ag-floating-bottom-viewport"
                  role="presentation"
                >
                  <div
                    class="ag-floating-bottom-container"
                    role="presentation"
                    style="width: 0px;"
                  />
                </div>
                <!-- AG Row Container bottomRight -->
                <div
                  class="ag-pinned-right-floating-bottom"
                  role="presentation"
                />
                <!-- AG Row Container bottomFullWidth -->
                <div
                  class="ag-floating-bottom-full-width-container"
                  role="presentation"
                />
              </div>
              <!-- AG Fake Horizontal Scroll -->
              <div
                aria-hidden="true"
                class="ag-body-horizontal-scroll"
              >
                
            
                <div
                  class="ag-horizontal-left-spacer"
                  ref="eLeftSpacer"
                  style="width: 0px; max-width: 0px; min-width: 0px;"
                />
                
            
                <div
                  class="ag-body-horizontal-scroll-viewport"
                  ref="eViewport"
                >
                  
                
                  <div
                    class="ag-body-horizontal-scroll-container"
                    ref="eContainer"
                    style="width: 0px;"
                  />
                  
            
                </div>
                
            
                <div
                  class="ag-horizontal-right-spacer"
                  ref="eRightSpacer"
                  style="width: 0px; max-width: 0px; min-width: 0px;"
                />
                
        
              </div>
              <!-- AG Overlay Wrapper -->
              <div
                aria-hidden="true"
                class="ag-overlay"
              >
                
            
                <div
                  class="ag-overlay-panel"
                >
                  
                
                  <div
                    class="ag-overlay-wrapper ag-layout-normal ag-overlay-loading-wrapper"
                    ref="eOverlayWrapper"
                  >
                    <span
                      class="ag-overlay-loading-center"
                    >
                      Loading...
                    </span>
                  </div>
                  
            
                </div>
                
        
              </div>
            </div>
            <div
              class="ag-tab-guard ag-tab-guard-bottom"
              role="presentation"
              tabindex="0"
            />
          </div>
          <div
            aria-hidden="true"
            class="ag-paging-panel ag-unselectable ag-hidden"
            id="ag-4"
          >
            
                
            <!--AG-PAGE-SIZE-SELECTOR-->
            <span
              class="ag-paging-page-size"
              ref="pageSizeComp"
            />
            
                
            <span
              class="ag-paging-row-summary-panel"
              role="status"
            >
              
                    
              <span
                class="ag-paging-row-summary-panel-number"
                id="ag-4-first-row"
                ref="lbFirstRowOnPage"
              />
              
                    
              <span
                id="ag-4-to"
              >
                to
              </span>
              
                    
              <span
                class="ag-paging-row-summary-panel-number"
                id="ag-4-last-row"
                ref="lbLastRowOnPage"
              />
              
                    
              <span
                id="ag-4-of"
              >
                of
              </span>
              
                    
              <span
                class="ag-paging-row-summary-panel-number"
                id="ag-4-row-count"
                ref="lbRecordCount"
              />
              
                
            </span>
            
                
            <span
              class="ag-paging-page-summary-panel"
              role="presentation"
            >
              
                    
              <div
                aria-label="First Page"
                class="ag-button ag-paging-button"
                ref="btFirst"
                role="button"
                tabindex="0"
              >
                <span
                  class="ag-icon ag-icon-first"
                  role="presentation"
                  unselectable="on"
                />
              </div>
              
                    
              <div
                aria-label="Previous Page"
                class="ag-button ag-paging-button"
                ref="btPrevious"
                role="button"
                tabindex="0"
              >
                <span
                  class="ag-icon ag-icon-previous"
                  role="presentation"
                  unselectable="on"
                />
              </div>
              
                    
              <span
                class="ag-paging-description"
                role="status"
              >
                
                        
                <span
                  id="ag-4-start-page"
                >
                  Page
                </span>
                
                        
                <span
                  class="ag-paging-number"
                  id="ag-4-start-page-number"
                  ref="lbCurrent"
                />
                
                        
                <span
                  id="ag-4-of-page"
                >
                  of
                </span>
                
                        
                <span
                  class="ag-paging-number"
                  id="ag-4-of-page-number"
                  ref="lbTotal"
                />
                
                    
              </span>
              
                    
              <div
                aria-label="Next Page"
                class="ag-button ag-paging-button"
                ref="btNext"
                role="button"
                tabindex="0"
              >
                <span
                  class="ag-icon ag-icon-next"
                  role="presentation"
                  unselectable="on"
                />
              </div>
              
                    
              <div
                aria-label="Last Page"
                class="ag-button ag-paging-button"
                ref="btLast"
                role="button"
                tabindex="0"
              >
                <span
                  class="ag-icon ag-icon-last"
                  role="presentation"
                  unselectable="on"
                />
              </div>
              
                
            </span>
            
            
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
